cdc/sink/mq/mq.go

```diff
@@ -197,6 +197,15 @@ func (k *mqSink) FlushRowChangedEvents(
 
 // bgFlushTs flush resolvedTs to workers and flush the mqProducer
 func (k *mqSink) bgFlushTs(ctx context.Context) error {
+	defer func() {
+		go func() {
+			// We must finish consuming the data here,
+			// otherwise it will cause the channel to not close properly.
+			for range k.resolvedBuffer.Out() {
+				// Do nothing. We do not care about the data.
+			}
+		}()
```

Comment:
Shall we wait the for loop done in the defer func?

Reply:
No, closing is also done asynchronously, as long as we can return the error and the upper level calls close, we will always finish consuming the data and close the goruntine in chann normally.

Reply:
What about moving the cleanup for loop to `mqSink.Close()`?

Reply:
Good idea. Makes sense! Thanks

Reply:
Done